### DISCUSSION TITLE
I have created JapanProvider.cs that complies with Japanese holiday laws, so could you adopt it?

### DISCUSSION THREAD

**Discussion by hisashi-kuwahara** [nager/Nager.Date#570] [General]
I created a code that correctly calculates Japanese holidays from 1955 to 2026.
There is also code with verified unit tests.

**Comment by tinohager** [comment]
Hi thanks for your message, sounds very interesting, what about the coming years?

**Comment by hisashi-kuwahara** [comment]
I think this logic can be used for several years unless there is a revision to the holiday law or a major event in the imperial family.
